### PR TITLE
MSTeams PowerShell scripts for Teams

### PR DESCRIPTION
**Pull request by Geetha63**
Adding 5 scripts
1.Add App to Teams
Script is to add TeamsApp to Teams by providing the parameters TeamId and AppName

2.Add App to Teams and Tenant
The script is to add TeamsApp to Teams or tenant by providing the parameters OwnerPrincipalName, AppName and selecting the input

3.Create New-CsTeamsMessagingPolicy
The New-CsTeamsMessagingPolicy script enables administrators to control user TeamsMessagingPolicy, determines whether a user is allowed to chat. Set this to TRUE to allow a user to chat across the private chat. Set this to FALSE to prohibit private chat

4.CreateTeamsalongwithAPPs
Create MicrosoftTeams along with APP for your tenant

5.DeleteTeamsWhereTeamMembersdontHaveGivenJobtitles
This script checks each Team member job title, if at least one job title does not match to given job titles, the script will delete those Teams. It will generate the ouput.csv file in the current folder and sent an email to deleted Team owners

This is a Graph API script, to execute the script user needs to create an Azure App and provide the necessary permissions

### CODE REVIEW

File: Add App to Teams/Add-TeamsAppInstallation.ps1 [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿$logfile = "C:\TeamsAppInstallationlog_$(get-date -format `"yyyyMMdd_hhmmsstt`").txt"
```
Comment (MartinRinas) [inline Add App to Teams/Add-TeamsAppInstallation.ps1]: Please add Synopsis block, i.e. from this template for inline help. https://github.com/TeamsAdminSamples/PowerShell/blob/main/z_Templates/New-PowerShell.ps1

File: Add App to Teams and Tenant/Add App to Teams or Tenant  using Graph API.ps1 [hunk 2]
```diff
@@ -0,0 +1,96 @@
+param(
```
Comment (MartinRinas) [inline Add App to Teams and Tenant/Add App to Teams or Tenant  using Graph API.ps1:3]: Please add Synopsis block, i.e. from this template for inline help. https://github.com/TeamsAdminSamples/PowerShell/blob/main/z_Templates/New-PowerShell.ps1

File: Create New-CsTeamsMessagingPolicy/New-CsTeamsMessagingPolicy.ps1 [hunk 3]
```diff
@@ -0,0 +1,18 @@
+$logfile = "C:\CreateNew-CsTeamsMessagingPolicylog_$(get-date -format `"yyyyMMdd_hhmmsstt`").txt"
```
Comment (MartinRinas) [inline Create New-CsTeamsMessagingPolicy/New-CsTeamsMessagingPolicy.ps1]: Please add Synopsis block, i.e. from this template for inline help. https://github.com/TeamsAdminSamples/PowerShell/blob/main/z_Templates/New-PowerShell.ps1

File: CreateTeamsalongwithAPPs/CreateTeamsAlongWithApps.ps1 [hunk 4]
```diff
@@ -0,0 +1,126 @@
+
```
Comment (MartinRinas) [inline CreateTeamsalongwithAPPs/CreateTeamsAlongWithApps.ps1:2]: Please add Synopsis block, i.e. from this template for inline help. https://github.com/TeamsAdminSamples/PowerShell/blob/main/z_Templates/New-PowerShell.ps1

File: DeleteTeamsWhereTeamMembersdontHaveGivenJobtitles/DeleteTeamsWhereTeamMembersdontHaveGivenJobtitles.ps1 [hunk 5]
```diff
@@ -0,0 +1,153 @@
+﻿#this script will Delete the all teams except given job titles match
```
Comment (MartinRinas) [inline DeleteTeamsWhereTeamMembersdontHaveGivenJobtitles/DeleteTeamsWhereTeamMembersdontHaveGivenJobtitles.ps1]: Please add Synopsis block, i.e. from this template for inline help. https://github.com/TeamsAdminSamples/PowerShell/blob/main/z_Templates/New-PowerShell.ps1

File: Add App to Teams/Add-TeamsAppInstallation.ps1 [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+$start = [system.datetime]::Now
+
+Connect-MicrosoftTeams
+$AppName=Read-Host "Please enter AppName"
```
Comment (MartinRinas) [inline Add App to Teams/Add-TeamsAppInstallation.ps1]: consider turning these into required parameters, PowerShell will prompt for them if not provided

File: Add App to Teams/Add-TeamsAppInstallation.ps1 [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+try{
+$App=Get-TeamsApp -DisplayName "$AppName"
+$APP
+}
+catch{
+$_.Exception.Message | out-file -Filepath $logfile -append
+}
+
+$AppId =$App.Id
+$TeamId= Read-Host "Please enter TeamId"
```
Comment (MartinRinas) [inline Add App to Teams/Add-TeamsAppInstallation.ps1]: same as with AppName, consider using required parameters

File: Add App to Teams and Tenant/Add App to Teams or Tenant  using Graph API.ps1 [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+      [Parameter(Mandatory=$true)][System.String]$OwnerPrincipalName,
+      [Parameter(Mandatory=$true)][System.String]$AppName,
+      [Parameter(Mandatory=$true)][System.String]$Tenantid,
+      [Parameter(Mandatory=$true)][System.String]$client_Id,
+      [Parameter(Mandatory=$true)][System.String]$Client_Secret
+      )
+      
+Connect-MicrosoftTeams
```
Comment (MartinRinas) [inline Add App to Teams and Tenant/Add App to Teams or Tenant  using Graph API.ps1:26]: Script has no logging, might be good to add debug log capabilities as its a multi-step process. https://github.com/TeamsAdminSamples/PowerShell/blob/main/z_Templates/New-PowerShell.ps1 as a read-to-use logging function that can easily integrated.

File: CreateTeamsalongwithAPPs/CreateTeamsAlongWithApps.ps1 [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+$logfile = "C:\log_$(get-date -format `"yyyyMMdd_hhmmsstt`").txt"
```
Comment (MartinRinas) [inline CreateTeamsalongwithAPPs/CreateTeamsAlongWithApps.ps1]: similar suggestion on logging, rather be more verbose in logging for long scripts as it helps admins to understand what happens and where things fail.

File: DeleteTeamsWhereTeamMembersdontHaveGivenJobtitles/DeleteTeamsWhereTeamMembersdontHaveGivenJobtitles.ps1 [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+#Provide job title phrases separate by , 
+#for best practice run the script quote delete cmdlet
+
+$logfile = "C:\log_$(get-date -format `"yyyyMMdd_hhmmsstt`").txt"
```
Comment (MartinRinas) [inline DeleteTeamsWhereTeamMembersdontHaveGivenJobtitles/DeleteTeamsWhereTeamMembersdontHaveGivenJobtitles.ps1]: same comment for logging, consider Write-VerboseLog from template or other mechanism to have more verbose logging. Also, please do not use any absolute paths like c:\something in the script but use relative paths like .\mylogfile.txt only.

**Review by MartinRinas — CHANGES_REQUESTED** [review]
suggested a few updates to logging and missing synopsis.

File: Add App to Teams and Tenant/Add App to Teams or Tenant  using Graph API.ps1 (hunk 8 above)
Comment (Geetha63) [reply Add App to Teams and Tenant/Add App to Teams or Tenant  using Graph API.ps1:26]: Added login to the script.

**Pull request merged**